eng/packages/http-client-csharp-mgmt/emitter/src/emitter.ts

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { EmitContext } from "@typespec/compiler";
+import { DecoratorInfo } from "@azure-tools/typespec-client-generator-core";
+
+import {
+  CodeModel,
+  CSharpEmitterOptions,
+  InputModelType,
+  setSDKContextOptions
+} from "@typespec/http-client-csharp";
+
+import { $onEmit as $onAzureEmit } from "@azure-typespec/http-client-csharp";
+import { azureSDKContextOptions } from "./sdk-context-options.js";
+import { calculateResourceTypeFromPath } from "./resource-type.js";
+
+const armResourceOperations = "Azure.ResourceManager.@armResourceOperations";
+const armResourceRead = "Azure.ResourceManager.@armResourceRead";
+const armResourceCreateOrUpdate =
+  "Azure.ResourceManager.@armResourceCreateOrUpdate";
+const singleton = "Azure.ResourceManager.@singleton";
+const resourceMetadata = "Azure.ClientGenerator.Core.@resourceSchema";
+
+export async function $onEmit(context: EmitContext<CSharpEmitterOptions>) {
+  context.options["generator-name"] ??= "MgmtClientGenerator";
+  context.options["update-code-model"] = updateCodeModel;
+  context.options["emitter-extension-path"] ??= import.meta.url;
+  setSDKContextOptions(azureSDKContextOptions);
```

Comment:
Since we are running $onEmit from azure emitter, but we are not setting the defaultSDKContextOption there, this get lost. Will need to put this into options as well to pass along to MTG emitter.

Reply:
https://github.com/microsoft/typespec/pull/6550